include/event_dbus_monitor.hpp

```diff
@@ -0,0 +1,641 @@
+{
+    registerHostStateChangeSignal();
+    registerBMCStateChangeSignal();
+    registerBootProgressChangeSignal();
+}
+
+static void registerVMIConfigChangeSignal()
+{
+    registerVMIIPEnabledPropChangeSignal();
+    registerVMIIPChangeSignal();
+    registerVMIEthIntfPropChangeSignal();
+}
+
+void registerPostCodeChangeSignal()
+{
+    BMCWEB_LOG_DEBUG("PostCode change signal - Register");
+
+    matchPostCodeChange = std::make_unique<sdbusplus::bus::match::match>(
+        *crow::connections::systemBus,
+        "type='signal',member='PropertiesChanged',interface='org.freedesktop."
+        "DBus.Properties',path='/xyz/openbmc_project/state/boot/raw0',"
+        "arg0='xyz.openbmc_project.State.Boot.Raw'",
+        postCodePropertyChange);
+}
+
+inline void dumpCreatedSignal(sdbusplus::message::message& msg)
+{
+    BMCWEB_LOG_DEBUG("Dump Created - match fired");
+
+    if (msg.is_method_error())
+    {
+        BMCWEB_LOG_ERROR("Dump Created signal error");
+        return;
+    }
+
+    std::string dumpType;
+    std::string dumpId;
+
+    dbus::utility::getNthStringFromPath(msg.get_path(), 3, dumpType);
+    dbus::utility::getNthStringFromPath(msg.get_path(), 5, dumpId);
+
+    boost::container::flat_map<std::string, std::variant<std::string, uint8_t>>
+        values;
+    std::string objName;
+    msg.read(objName, values);
+
+    auto find = values.find("Status");
+    if (find == values.end())
+    {
+        BMCWEB_LOG_DEBUG("Status property not found. Continuing to listen...");
+        return;
+    }
+    std::string* propValue = std::get_if<std::string>(&(find->second));
+
+    if (propValue != nullptr &&
+        *propValue ==
+            "xyz.openbmc_project.Common.Progress.OperationStatus.Completed")
+    {
+        BMCWEB_LOG_DEBUG << "Sending event\n";
+
+        std::string eventOrigin;
+        // Push an event
+        if (dumpType == "bmc")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/{}", dumpId);
+        }
+        else if (dumpType == "system")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/System_{}",
+                dumpId);
+        }
+        else if (dumpType == "resource")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/Resource_{}",
+                dumpId);
+        }
+        else if (dumpType == "hostboot")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/Hostboot_{}",
+                dumpId);
+        }
+        else if (dumpType == "hardware")
+        {
+            eventOrigin = std::format(
+                "/redfish/v1/Systems/system/LogServices/Dump/Entries/Hardware_{}",
+                dumpId);
+        }
+        else if (dumpType == "sbe")
+        {
+            eventOrigin = std::format(
```

Comment:
Even if this may be the same with `boost::urls::format()` here, but it is better to use `boost::urls::format()`'.
(In some cases e.g. like `Assembly#/...` becomes different).

Reply:
Here is the good explanation about `boost::urls::format()`  - https://github.com/boostorg/url/blob/fddf6b75d5fffe45c016a0787cc951bda779772b/include/boost/url/format.hpp#L22

bmcweb suggests to use boost::urls::format() for the places that are dealing with URLs.  This would prevent any potential mistakes or future changes which may become inconsistent with std::format().

Anyway, it is okay for downstream, but this would be requested to change when doing upstream.